frontend/src/pages/storageClasses/StorageClassesTableRow.tsx

```diff
@@ -0,0 +1,194 @@
+import { updateStorageClassConfig } from '~/services/StorageClassService';
+import DashboardPopupIconButton from '~/concepts/dashboard/DashboardPopupIconButton';
+import { NoValue } from '~/components/NoValue';
+import { ColumnLabel } from './constants';
+import { isOpenshiftDefaultStorageClass } from './utils';
+import { StorageClassEnableSwitch } from './StorageClassEnableSwitch';
+import { StorageClassDefaultRadio } from './StorageClassDefaultRadio';
+
+interface StorageClassesTableRowProps {
+  storageClass: StorageClassKind;
+  storageClassConfigMap: Record<string, StorageClassConfig | undefined>;
+  refresh: FetchStateRefreshPromise<StorageClassKind[]>;
+}
+
+export const StorageClassesTableRow: React.FC<StorageClassesTableRowProps> = ({
+  storageClass,
+  storageClassConfigMap,
+  refresh,
+}) => {
+  const storageClassConfig = storageClassConfigMap[storageClass.metadata.name];
+  const isOpenshiftDefault = isOpenshiftDefaultStorageClass(storageClass);
+  const { metadata, provisioner, reclaimPolicy, volumeBindingMode, allowVolumeExpansion } =
+    storageClass;
+
+  const storageClassInfoItems = [
+    {
+      label: 'OpenShift storage class',
+      value: storageClassConfig?.displayName,
+    },
+    {
+      label: 'Provisioner',
+      value: provisioner,
+    },
+    {
+      label: 'Description',
+      value: storageClassConfig?.description ?? <NoValue />,
+    },
+    {
+      label: 'Reclaim policy',
+      value: reclaimPolicy,
+    },
+    {
+      label: 'Volume binding mode',
+      value: volumeBindingMode,
+    },
+    {
+      label: 'Allow PersistentVolumeClaims to be expanded',
+      value: allowVolumeExpansion ? 'True' : 'False',
+    },
+  ];
+
+  const isEnableSwitchDisabled = React.useMemo(() => {
+    const hasOtherStorageClassesEnabled = Object.entries(storageClassConfigMap).some(
+      ([storageClassName, config]) =>
+        storageClassName !== storageClass.metadata.name && config?.isEnabled,
+    );
+
+    // Prevent disabling when the storage class is enabled and all
+    // others are disabled or if the storage class is set as the default
+    if (
+      storageClassConfig?.isEnabled &&
+      (!hasOtherStorageClassesEnabled || storageClassConfig.isDefault)
+    ) {
+      return true;
+    }
+
+    return false;
+  }, [
+    storageClass.metadata.name,
+    storageClassConfig?.isDefault,
+    storageClassConfig?.isEnabled,
+    storageClassConfigMap,
+  ]);
+
+  const onDefaultRadioChange = React.useCallback(async () => {
+    const existingDefaultConfigMap = Object.entries(storageClassConfigMap).find(
+      ([name, config]) => storageClass.metadata.name !== name && config?.isDefault,
+    );
+
+    if (existingDefaultConfigMap) {
+      const [name, config] = existingDefaultConfigMap;
+      await updateStorageClassConfig(name, { ...config, isDefault: false });
+      refresh();
+    }
+  }, [storageClass.metadata.name, storageClassConfigMap, refresh]);
```

Comment:
Is `storageClassConfigMap` reference stable here?

Reply:
Updated